archunit/src/main/java/com/tngtech/archunit/lang/conditions/ArchConditions.java

```diff
@@ -1205,6 +1209,35 @@ public static AllDependenciesCondition onlyHaveDependenciesWhere(DescribedPredic
         return new DoesConditionByPredicate<>(declareThrowableOfType);
     }
 
+    @PublicAPI(usage = ACCESS)
+    public static ArchCondition<JavaCodeUnit> onlyBeCalledByClassesThat(DescribedPredicate<? super JavaClass> predicate) {
+        ChainableFunction<JavaAccess<?>, JavaCodeUnit> origin = JavaAccess.Functions.Get.origin();
+        ChainableFunction<HasOwner<JavaClass>, JavaClass> owner = Get.owner();
+        return new CodeUnitCallsCondition(Joiner.on(" ").join("only be called by classes that", predicate.getDescription()),
```

Comment:
> Wouldn't it be simpler to just write `"only be called by classes that " + predicate.getDescription()`?
Yes, it's simpler - changed :smiley:

> But if we unify `CodeUnitCallsCondition` it would likely just take description and predicate into the constructor separately anyway 🤷‍♂️ (at least that's how the respective `Classes...` does it)
I've left it as it is for now as I'm not sure how to change it. There are 4 methods in _ArchConditions_ using the same object _CodeUnitOnlyCallsCondition_, each with different description (i.e. "only be called by classes that “, “only be called by methods that "), that is why I'm passing description here. Also I’m passing extended predicate (i.e. origin.is(method().and(predicate)) so extracting predicate description inside _CodeUnitOnlyCallsCondition_ constructor would end up with description: “… should only be called by methods that is method and custom predicate”.

Reply:
Yeah, sorry, I got confused there :upside_down_face: It's already exactly how the other conditions do it, so never mind :slightly_smiling_face: